File: 2019/test/day01_test.clj
(ns day01-test
  (:require [day01 :as sut]
            [clojure.test :as t]))

(mapv sut/part1
      [12 14 1969 100756]) ; => [2 2 654 33583]

(sut/part2 14) ; => 2
(sut/part2 1969) ; => 966
(sut/part2 100756) ; => 50346

File: 2023/python/day01_test.py
from day01 import part1, part2

IN = open('day01_input.txt').readlines()


TEST = """
1abc2
pqr3stu8vwx
a1b2c3d4e5f
treb7uchet
""".strip().splitlines()

print('part1 test:', part1(TEST))
print('part1:', part1(IN))

TEST = """
two1nine
eightwothree
abcone2threexyz
xtwone3four
4nineeightseven2
zoneight234
7pqrstsixteen
""".strip().splitlines()

print('part2 test:', part2(TEST))
print('part2:', part2(IN))
